packages/beacon-node/src/chain/opPools/aggregatedAttestationPool.ts

```diff
@@ -421,6 +551,34 @@ export function aggregateInto(attestation1: AttestationWithIndex, attestation2:
   attestation1.attestation.signature = bls.Signature.aggregate([signature1, signature2]).toBytes();
 }
 
+/**
+ * Electra and after: Block proposer consolidates attestations with the same
+ * attestation data from different committee into a single attestation
+ * https://github.com/ethereum/consensus-specs/blob/aba6345776aa876dad368cab27fbbb23fae20455/specs/_features/eip7549/validator.md?plain=1#L39
+ */
+export function aggregateConsolidation({byCommittee, attData}: AttestationsConsolidation): electra.Attestation {
+  const committeeBits = BitArray.fromBitLen(MAX_COMMITTEES_PER_SLOT);
+  // TODO: can we improve this?
+  let aggregationBits: boolean[] = [];
+  const signatures: Signature[] = [];
+  const sortedCommittees = Array.from(byCommittee.keys()).sort((a, b) => a - b);
+  for (const committeeIndex of sortedCommittees) {
+    const attestationNonParticipation = byCommittee.get(committeeIndex);
+    if (attestationNonParticipation !== undefined) {
+      const {attestation} = attestationNonParticipation;
+      committeeBits.set(committeeIndex, true);
+      aggregationBits = [...aggregationBits, ...attestation.aggregationBits.toBoolArray()];
```

Comment:
Chain up aggregationBits by using `toBoolArray()` and later do `fromBoolArray()` seems like a hack. Looks good for now but this chain up operation should ideally be provided by ssz library

Reply:
hmm like a concat op